packages/tools/client-debugger/devtools/src/index.ts

```diff
@@ -0,0 +1,238 @@
+	 *
+	 * If not provided, the {@link ContainerDevtoolsProps.containerId} will be used for the purpose of distinguishing
+	 * instances.
+	 */
+	containerNickname?: string;
+
+	/**
+	 * (optional) Configurations for generating visual representations of
+	 * {@link @fluidframework/shared-object-base#ISharedObject}s under each Container's
+	 * {@link @fluidframework/fluid-static#IFluidContainer.initialObjects}.
+	 *
+	 * @remarks
+	 *
+	 * If not specified, then only `SharedObject` types natively known by the system will be visualized, and using
+	 * default visualization implementations.
+	 *
+	 * Any visualizer configurations specified here will take precedence over system defaults, as well as any
+	 * provided when initializing the Devtools.
+	 */
+	dataVisualizers?: Record<string, VisualizeSharedObject>;
+}
+
+/**
+ * Fluid Devtools. A single, global instance is used to generate and communicate stats associated with the general Fluid
+ * runtime (i.e., it is not associated with any single Framework entity).
+ *
+ * @remarks
+ *
+ * Supports registering {@link @fluidframework/fluid-static#IFluidContainer}s for Container-level stats
+ * (via {@link IDevtools.registerContainerDevtools}).
+ *
+ * The lifetime of the associated singleton is bound by that of the Window (globalThis), and it will be automatically
+ * disposed of on Window unload.
+ * If you wish to dispose of it earlier, you may call its {@link @fluidframework/common-definitions#IDisposable.dispose} method.
+ *
+ * @public
+ */
+export interface IDevtools extends IDisposable {
+	/**
+	 * Initializes a {@link IDevtools} from the provided properties and stores it for future reference.
+	 *
+	 * @throws Will throw if devtools have already been registered for the specified Container ID.
+	 */
+	registerContainerDevtools(props: ContainerDevtoolsProps): void;
+
+	/**
+	 * Closes registered Container-level Devtools associated with the provided ID.
+	 */
+	closeContainerDevtools(id: string): void;
+}
+
+class Devtools implements IDevtools {
+	public constructor(
+		/**
+		 * Handle to the underlying Devtools instance (singleton).
+		 */
+		private readonly _devtools: IDevtoolsBase,
+	) {}
+
+	/**
+	 * {@inheritDoc IDevtools.registerContainerDevtools}
+	 */
+	public registerContainerDevtools(props: ContainerDevtoolsProps): void {
+		const mappedProps = mapContainerProps(props);
+		if (mappedProps !== undefined) {
+			this._devtools.registerContainerDevtools(mappedProps);
+		}
+	}
+
+	/**
+	 * {@inheritDoc IDevtools.closeContainerDevtools}
+	 */
+	public closeContainerDevtools(id: string): void {
+		this._devtools.closeContainerDevtools(id);
+	}
+
+	/**
+	 * {@inheritDoc IDevtools.disposed}
+	 */
+	public get disposed(): boolean {
+		return this._devtools.disposed;
+	}
+
+	/**
+	 * {@inheritDoc IDevtools.dispose}
+	 */
+	public dispose(): void {
+		this._devtools.dispose();
+	}
+}
+
+/**
+ * TODO
+ *
+ * @public
+ */
+export function initializeDevtools(props: DevtoolsProps): IDevtools {
```

Comment:
Non-blocking feedback: I'm a bit conflicted that the functions are called the same as in devtools-core but don't have great ideas about it.

Reply:
I see the two entry-points as being mutually exclusive (one just happens to use the other as an implementation detail). So hopefully it won't lead to any confusion, but we can definitely chat about this more.